test/gateway/eip20_gateway/confirm_redeem_intent.js

```diff
@@ -0,0 +1,539 @@
+    );
+
+    assert.strictEqual(
+      tx.receipt.status,
+      true,
+      'Transaction should success.',
+    );
+
+  }
+
+  beforeEach(async function () {
+
+      mockToken = await MockToken.new({from: accounts[0]});
+      baseToken = await MockToken.new({from: accounts[0]});
+
+      redeemRequest = TestDataJSON.gateway.confirm_redeem_intent.params;
+      redeemRequest.nonce = new BN(redeemRequest.nonce, 16);
+      redeemRequest.amount = new BN(redeemRequest.amount, 16);
+      redeemRequest.gasLimit = new BN(redeemRequest.gasLimit, 16);
+      redeemRequest.blockNumber = new BN(redeemRequest.blockNumber, 16);
+      redeemRequest.gasPrice = new BN(redeemRequest.gasPrice, 16);
+
+      owner = accounts[4];
+      worker = accounts[8];
+
+      mockOrganization = await MockOrganization.new(owner, worker);
+
+      utilityToken = await Token.new(
+        accounts[3],
+        "",
+        "",
+        18,
+        owner,
+        {from: accounts[0]},
+      );
+
+      coreAddress = accounts[3];
+      burnerAddress = accounts[6];
+      bountyAmount = new BN(100);
+      gatewayAddress = testData;
+      dummyStateRootProvider = accounts[2];
+
+      eip20Gateway = await EIP20Gateway.new(
+        mockToken.address,
+        baseToken.address,
+        dummyStateRootProvider,
+        bountyAmount,
+        mockOrganization.address,
+        burnerAddress,
+      );
+
+      gatewayUtils = new GatewayUtils(eip20Gateway, mockToken, baseToken);
+      await eip20Gateway.activateGateway(TestDataJSON.contracts.coGateway, {from: owner});
+      await eip20Gateway.setStorageRoot(
+        redeemRequest.blockNumber,
+        redeemRequest.storageRoot,
+      );
+
+    }
+  );
+
+  it('should pass when all the params are valid', async function () {
+
+      let result = await eip20Gateway.confirmRedeemIntent.call(
+        redeemRequest.redeemer,
+        redeemRequest.nonce,
+        redeemRequest.beneficiary,
+        redeemRequest.amount,
+        redeemRequest.gasPrice,
+        redeemRequest.gasLimit,
+        redeemRequest.blockNumber,
+        redeemRequest.hashLock,
+        redeemRequest.storageProof,
+      );
+
+      let messageHash = TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_;
+      assert.strictEqual(
+        result,
+        messageHash,
+        `Message hash from event must be equal to ${messageHash}.`,
+      );
+
+      let tx = await eip20Gateway.confirmRedeemIntent(
+        redeemRequest.redeemer,
+        redeemRequest.nonce,
+        redeemRequest.beneficiary,
+        redeemRequest.amount,
+        redeemRequest.gasPrice,
+        redeemRequest.gasLimit,
+        redeemRequest.blockNumber,
+        redeemRequest.hashLock,
+        redeemRequest.storageProof,
+      );
+
+      assert.strictEqual(
+        tx.receipt.status,
+        true,
+        'Transaction should success.',
+      );
+
+    }
+  );
+
+  it('should pass when redeemer address is zero', async function () {
```

Comment:
```suggestion
  it('should fail when redeemer address is zero', async function () {
```

Please fix across the tests